en/docs/develop/mi-for-vscode/best-practices.md

```diff
@@ -0,0 +1,234 @@
+        | Comments          | *Enter Comments*            |
+    
+    - **Test Summary Report**: Prepare a summary document that aggregates all test results. Include the total number of test cases executed, breakdown of passed/failed cases, key observations, and a concise summary of overall system quality and readiness.
+    - **Traceability**: Link each test result to the corresponding requirement or user story to ensure traceability and coverage.
+    - **Review & Sign-off**: Share the test result documents and summary with stakeholders for review and sign-off before proceeding to deployment.
+
+## Deployment Best Practices
+
+- Before deploying to production, optimize your deployment environment based on WSO2 MI performance recommendations. Review and implement the [production deployment guidelines](https://mi.docs.wso2.com/en/latest/install-and-setup/setup/deployment-best-practices/production-deployment-guidelines/) to ensure scalability, reliability, and security. Adjust resource allocations, configure JVM parameters, and fine-tune network and storage settings as outlined in the official documentation.
+
+- **Pre-testing Patches**: Always test patches in a dedicated test environment before applying them to production. This helps identify potential issues and ensures compatibility with your integration solutions.
+
+- **Automated Patch Testing**: Use automated test suites such as JMeter or SoapUI to validate the stability and functionality of patched environments. Automate regression and integration tests to quickly detect any breaking changes.
+
+- **Patch Documentation**: Document all patches applied, including version details, test results, and any issues encountered during testing. Maintain a changelog for traceability and future reference.
+
+- **Scheduled Patch Cycles**: Establish a regular patching schedule and communicate planned maintenance windows to stakeholders. Avoid applying patches during peak business hours to minimize disruption.
+
+- **Rollback Strategy**: Prepare a rollback plan in case a patch introduces unexpected issues. Ensure backups and previous versions are available for quick restoration.
+
+- **Security Patches**: Prioritize security patches and apply them promptly to mitigate vulnerabilities. Validate security fixes through targeted penetration and vulnerability tests in the test environment.
+
+## Security Best Practices
+
+- Use [WSO2 Secret Vault](https://mi.docs.wso2.com/en/latest/install-and-setup/setup/security/encrypting-plain-text/) to securely store and manage sensitive information such as API keys, passwords, and certificates in your integration solutions. This approach ensures that secrets are encrypted at rest and only decrypted at runtime, reducing the risk of unauthorized access and enhancing overall security compliance.
+
+## Migration Best Practices
+
+If you are migrating from an older version of WSO2 Micro Integrator (&lt;4.4.0), consider the following best practices:
```

Comment:
Shall we mention these are still working, but consider modifying the artifacts to align with the new semantics.

Reply:
resolved in [bc39d38](https://github.com/wso2/docs-mi/pull/1789/commits/bc39d38f53345b6fcde9f701464af15698becdc5)